Library/DiscUtils.Dmg/UdifBuffer.cs

```diff
@@ -91,7 +91,7 @@ public override int Read(long pos, byte[] buffer, int offset, int count)
 
                     case RunType.Raw:
                         _stream.Position = _activeRun.CompOffset + bufferOffset;
-                        StreamUtilities.ReadFully(_stream, buffer, offset + totalCopied, toCopy);
+                        StreamUtilities.ReadExact(_stream, buffer, offset + totalCopied, toCopy);
```

Comment:
Did not throw, does now. Full read is assumed, so correct behavior is to throw on short read.